includes/class-wc-order-item-product.php

```diff
@@ -391,73 +219,235 @@ public function get_type() {
 	/**
 	 * Get subtotal.
+	 *
+	 * @param  string $context
 	 * @return string
 	 */
-	public function get_subtotal() {
-		return wc_format_decimal( $this->_data['subtotal'] );
+	public function get_subtotal( $context = 'view' ) {
+		return $this->get_prop( 'subtotal', $context );
 	}
 
 	/**
 	 * Get subtotal tax.
+	 *
+	 * @param  string $context
 	 * @return string
 	 */
-	public function get_subtotal_tax() {
-		return wc_format_decimal( $this->_data['subtotal_tax'] );
+	public function get_subtotal_tax( $context = 'view' ) {
+		return $this->get_prop( 'subtotal_tax', $context );
 	}
 
 	/**
 	 * Get total.
+	 *
+	 * @param  string $context
 	 * @return string
 	 */
-	public function get_total() {
-		return wc_format_decimal( $this->_data['total'] );
+	public function get_total( $context = 'view' ) {
+		return $this->get_prop( 'total', $context );
 	}
 
 	/**
 	 * Get total tax.
+	 *
+	 * @param  string $context
 	 * @return string
 	 */
-	public function get_total_tax() {
-		return wc_format_decimal( $this->_data['total_tax'] );
+	public function get_total_tax( $context = 'view' ) {
+		return $this->get_prop( 'total_tax', $context );
 	}
 
 	/**
 	 * Get fee taxes.
+	 *
+	 * @param  string $context
 	 * @return array
 	 */
-	public function get_taxes() {
-		return $this->_data['taxes'];
+	public function get_taxes( $context = 'view' ) {
+		return $this->get_prop( 'taxes', $context );
+	}
+
+	/**
+	 * Get the associated product.
+	 *
+	 * @param  string $context
+	 * @return WC_Product|bool
+	 */
+	public function get_product() {
+		if ( $this->get_variation_id() ) {
+			$product = wc_get_product( $this->get_variation_id() );
+		} else {
+			$product = wc_get_product( $this->get_product_id() );
+		}
+
+		// Backwards compatible filter from WC_Order::get_product_from_item()
+		if ( has_filter( 'woocommerce_get_product_from_item' ) ) {
+			$product = apply_filters( 'woocommerce_get_product_from_item', $product, $this, wc_get_order( $this->get_order_id() ) );
+		}
+
+		return apply_filters( 'woocommerce_order_item_product', $product, $this );
+	}
+
+	/**
+	 * Get the Download URL.
+	 *
+	 * @param  int $download_id
+	 * @return string
+	 */
+	public function get_item_download_url( $download_id ) {
+		$order = $this->get_order();
+
+		return $order ? add_query_arg( array(
+			'download_file' => $this->get_variation_id() ? $this->get_variation_id() : $this->get_product_id(),
+			'order'         => $order->get_order_key(),
+			'email'         => urlencode( $order->get_billing_email() ),
+			'key'           => $download_id,
+		), trailingslashit( home_url() ) ) : '';
+	}
+
+	/**
+	 * Get any associated downloadable files.
+	 *
+	 * @return array
+	 */
+	public function get_item_downloads() {
+		$files   = array();
+		$product = $this->get_product();
+		$order   = $this->get_order();
+		if ( $product && $order && $product->is_downloadable() && $order->is_download_permitted() ) {
+			$download_ids = $this->data_store->get_download_ids( $this, $order );
```

Comment:
Might be a change after order data PR is in - I added a custom download data store.